File: frontend/src/types/triage.ts
import type { Pagination } from './person';

export interface Triage {
  id: string;
  person_id: string;
  campus_id: string;
  campaign_id?: string;
  main_complaint: string;
  assigned_team?: string;
  triage_date: string;
  location?: string;
  triaged_by: string;
  notes?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  requested_service_types?: string[];
}

export interface TriageListItem {
  id: string;
  person_id: string;
  person_name: string;
  main_complaint: string;
  triage_date: string;
  requested_service_count: number;
}

export interface TriageListResponse {
  data: TriageListItem[];
  pagination: Pagination;
}

export interface CreateTriageInput {
  person_id: string;
  campaign_id?: string;
  main_complaint: string;
  assigned_team?: string;
  triage_date?: string;
  location?: string;
  notes?: string;
  requested_service_types?: string[];
}

// The campaign link is set at creation only; the backend update input does
// not accept campaign_id, so the update type excludes it.
export type UpdateTriageInput = Omit<CreateTriageInput, 'person_id' | 'campaign_id'>;

File: backend/internal/domain/triage.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// Triage represents an initial intake assessment for an assisted person.
type Triage struct {
	ID             uuid.UUID   `json:"id"`
	PersonID       uuid.UUID   `json:"person_id"`
	CampaignID     *uuid.UUID  `json:"campaign_id,omitempty"`
	CampusID       uuid.UUID   `json:"campus_id"`
	MainComplaint  string      `json:"main_complaint"`
	AssignedTeam   *uuid.UUID  `json:"assigned_team,omitempty"`
	TriageDate     time.Time   `json:"triage_date"`
	Location       *string     `json:"location,omitempty"`
	TriagedBy      uuid.UUID   `json:"triaged_by"`
	Notes          *string     `json:"notes,omitempty"`
	IsActive       bool        `json:"is_active"`
	CreatedAt      time.Time   `json:"created_at"`
	UpdatedAt      time.Time   `json:"updated_at"`
	RequestedTypes []uuid.UUID `json:"requested_service_types,omitempty"`
}

// TriageListItem is a summary representation for list responses.
type TriageListItem struct {
	ID             uuid.UUID `json:"id"`
	PersonID       uuid.UUID `json:"person_id"`
	PersonName     string    `json:"person_name"`
	MainComplaint  string    `json:"main_complaint"`
	TriageDate     time.Time `json:"triage_date"`
	RequestedCount int       `json:"requested_service_count"`
}

// TriageListResult contains paginated triage list data.
type TriageListResult struct {
	Data       []TriageListItem `json:"data"`
	Pagination Pagination       `json:"pagination"`
}

// TriageFilter defines search/filter criteria for listing triages.
type TriageFilter struct {
	CampusID uuid.UUID
	PersonID *uuid.UUID
	From     *time.Time
	To       *time.Time
	Page     int
	PerPage  int
}
